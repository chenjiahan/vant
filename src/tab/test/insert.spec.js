import { mount, later } from '../../../test';

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab dynamically', async () => {
  const wrapper = mount({
    template: `
      <van-tabs v-model="active">
        <van-tab title="1">1</van-tab>
        <div v-if="insert">
          <van-tab title="2">2</van-tab>
        </div>
        <van-tab title="3">3</van-tab>
      </van-tabs>
    `,
    data() {
      return {
        insert: false,
        active: 1,
      };
    },
  });

  await later();
  wrapper.setData({ insert: true });
  expect(wrapper).toMatchSnapshot();
});
