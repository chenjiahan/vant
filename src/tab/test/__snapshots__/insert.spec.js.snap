// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert tab dynamically 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" class="van-tab van-ellipsis"><span class="van-tab__text">1<!----></span></div>
      <div role="tab" aria-selected="true" class="van-tab van-tab--active van-ellipsis"><span class="van-tab__text">2<!----></span></div>
      <div role="tab" class="van-tab van-ellipsis"><span class="van-tab__text">3<!----></span></div>
      <div class="van-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div>
      <div role="tabpanel" class="van-tab__pane">2</div>
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">3</div>
  </div>
</div>
`;

exports[`insert tab with child component 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active van-ellipsis"><span class="van-tab__text">1<!----></span></div>
      <div role="tab" class="van-tab van-ellipsis"><span class="van-tab__text">2<!----></span></div>
      <div role="tab" class="van-tab van-ellipsis"><span class="van-tab__text">3<!----></span></div>
      <div class="van-tabs__line"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">1</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;
